pkg/urbit/vere/pier.c

```diff
@@ -979,8 +1033,8 @@ _pier_boot_vent(u3_pier* pir_u)
       //
       //    XX this is horrible
       //
-      u3_noun tuf = ( c3__fake == u3h(pir_u->bot) ) ? u3_nul :
-                    u3h(u3t(u3t(u3t(u3t(pir_u->bot)))));
+      u3_noun tuf = ( c3y == pir_u->fak_o ) ? u3_nul :
```

Comment:
Can you document what this horrible thing is doing, briefly? You seem to understand it at least

Reply:
Working around vane initialization problems be extracting the ames domains from the boot event, if we're not fake, in order to have the i/o driver ready for any packets that are sent during the boot event. I'm separately working on making ames more resilient to the initialization order so we don't have to do this.

Reply:
Ok. A comment to this effect would be helpful.